### PR TITLE
Refact0r

### PR DESCRIPTION
**Pull request by sullof**
@Kondax 
I am proposing a simple refactoring of the way we inject a Promise in methods.
It reduces the risk of errors because any injection requires only two parameters, the method and the `arguments`. Like
```
return this.injectPromise(this.getTransactionsRelated, arguments);
```
I tested it in TronBox and it seems to work well. Let me know what you think of it?

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -199,7 +200,7 @@ export default class TronWeb extends EventEmitter {
 
     getEventByTransactionID(transactionID = false, callback = false) {
         if(!callback)
-            return this.injectPromise(this.getEventByTransactionID, transactionID);
+            return this.injectPromise(this.getEventByTransacionID, arguments);
```
Comment (Kondax) [inline src/index.js]: Typo, should be `getEventByTransactionID`

**Comment by Kondax** [comment]
I'm happy with using `arguments` as opposed to each one individually. One issue - shouldn't we shorten the arguments array by one? If callback is false, passing the callback in again would just result in an infinite loop?

Also, I don't think extending a class with only static methods has any advantage. Sticking with the old implementation (this.injectPromise = x) is a cleaner solution in my opinion.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
         "example": "node ./examples/server",
         "test:browser": "node scripts/test-browser.js && npx karma start --single-run --browsers ChromeHeadless,Firefox,Edge",
         "test:node": "node scripts/test-node.js && npx mocha ./test/*.js",
-        "test": "npm run-script test:browser && npm run-script test:node"
+        "test": "npm run-script test:browser && npm run-script test:node",
+        "build:dev": "NODE_ENV=development yarn build"
```
Comment (Kondax) [inline package.json]: Fairly certain this won't work in windows, if you look at the webpack docs I believe you can pass an argument for it to build a development version

File: src/utils/index.js [hunk 3]
```diff
@@ -67,22 +67,6 @@ const utils = {
...
-            func(...args, (err, res) => {
-                if(err)
-                    reject(err);
-                else resolve(res);
-            });
-        });
-    },
-
-    promiseInjector(scope) {
-        return (func, ...args) => {
```
Comment (Kondax) [inline src/utils/index.js]: You can change this to `return (func, args) => {`

File: src/utils/index.js [hunk 4]
```diff
@@ -67,22 +67,6 @@ const utils = {
...
-            return this.injectPromise(func.bind(scope), ...args);
```
Comment (Kondax) [inline src/utils/index.js]: And this to `return this.injectPromise(func.bind(scope), [ ...args ]);`.

This should support your `arguments` implementation (then just slice the args array in `injectPromise` to remove the last callback).

File: src/index.js (hunk 1 above)
Comment (sullof) [reply src/index.js]: Right. That is a typo. Thanks for catching it.

**Comment by sullof** [comment]
> Sticking with the old implementation (this.injectPromise = x) is a cleaner solution in my opinion.

I agree. I started with the Promisable class realizing after that I was not really using the class.

**Comment by sullof** [comment]
Anyway, for now I will close it. When I have a moment I will change it, if it makes sense.

**Pull request closed without merging**